static/js/invite.js

```diff
@@ -2,31 +2,42 @@ var invite = (function () {
 
 var exports = {};
 
-function update_subscription_checkboxes() {
-    // TODO: If we were more clever, we would only do this if the
-    // stream list has actually changed; that way, the settings of the
-    // checkboxes are saved from invocation to invocation (which is
-    // nice if I want to invite a bunch of people at once)
+// `get_invite_streams` is further modification of stream_data.invite_streams(), it is
+// defined here to keep stream_data.invite_stream() generic.
+exports.get_invite_streams = function () {
     var streams = [];
 
     _.each(stream_data.invite_streams(), function (value) {
+        var is_invite_only = stream_data.get_invite_only(value);
         var is_notifications_stream = value === page_params.notifications_stream;
+
+        // You can't actually elect to invite someone to the
+        // notifications stream. We won't even show it as a choice unless
+        // it's the only stream you have, or if you've made it private.
         if (stream_data.subscribed_streams().length === 1 ||
             !is_notifications_stream ||
-            is_notifications_stream && stream_data.get_invite_only(value)) {
-            // You can't actually elect not to invite someone to the
-            // notifications stream. We won't even show it as a choice unless
-            // it's the only stream you have, or if you've made it private.
-            var default_status = stream_data.get_default_status(value);
-            var invite_status = stream_data.get_invite_only(value);
-            streams.push({name: value, invite_only: invite_status, default_stream: default_status});
+            is_notifications_stream && is_invite_only) {
+
+            streams.push({
+                name: value,
+                invite_only: is_invite_only,
+                default_stream: stream_data.get_default_status(value),
+            });
+
             // Sort by default status.
             streams.sort(function (a, b) {
                 return b.default_stream - a.default_stream;
             });
         }
     });
-    $('#streams_to_add').html(templates.render('invite_subscription', {streams: streams}));
+
+    return streams;
+};
+
+function update_subscription_checkboxes() {
+    $('#streams_to_add').html(templates.render('invite_subscription', {
+        streams: exports.get_invite_streams(),
+    }));
```

Comment:
nitpick: I prefer a style where you can read the code in the order it happens:

~~~ js
var data = {stream: exports.get_invite_streams()};
var html = template.render('invite_subscription');
$('#streams_to_add').html(html)
~~~

Reply:
yeah, this is way much better, just confused as once I was asked to not prefer variable if used only once, but it makes sense here to use otherwise the line becomes long... so changing it to this